context/deletionallowedcontext/deletion_allowed_context.go

```diff
@@ -0,0 +1,62 @@
+// Package deletionallowedcontext stores and accesses the deletion allowed in
+// context.Context.
+package deletionallowedcontext
+
+import (
+	"context"
+)
+
+// key is an unexported type for keys defined in this package. This prevents
+// collisions with keys defined in other packages.
+type key string
+
+// deletionAllowedKey is the key for deletion allowed values in context.Context.
+// Clients use deletionallowedcontext.NewContext and
+// deletionallowedcontext.FromContext instead of using this key directly.
+var deletionAllowedKey key = "deletionallowed"
+
+// NewContext returns a new context.Context that carries value v.
+func NewContext(ctx context.Context, v chan struct{}) context.Context {
```

Comment:
Does `v` has to be parameter here? Why it can't be created inside `NewContext` as an "internal"?

Reply:
No reason. Just copy/paste from the original golang example to follow the language convention.